greenplum/cluster.go

```diff
@@ -351,8 +351,9 @@ func runStartStopCmd(stream step.OutStreams, gphome, command string, env string)
 // IsMasterRunning returns whether the cluster's master process is running.
 func (c *Cluster) IsMasterRunning(stream step.OutStreams) (bool, error) {
 	path := filepath.Join(c.MasterDataDir(), "postmaster.pid")
-	if !upgrade.PathExists(path) {
-		return false, nil
+	exist, err := upgrade.PathExist(path)
+	if !exist {
+		return false, err
```

Comment:
This needs to check the err from PathExists!

```suggestion
	exist, err := upgrade.PathExist(path)
	if err != nil {
		return false, err	
	}

	if !exist {
		return false, err
```

Reply:
If there is an error in the `PathExists` call, `exist` will be set to `false`. The first if check is already covered by the second one since they return the exact same pair.

Reply:
I'd much rather be verbose and extra clear in this case as it prevents people from needing to think through things like that.